chips/qemu_rv32_virt_chip/src/chip.rs

```diff
@@ -0,0 +1,279 @@
+//! High-level setup and interrupt mapping for the chip.
+
+use core::fmt::Write;
+
+use capsules::virtio::interfaces::mmio::VirtIOMMIODevice;
+use kernel;
+use kernel::debug;
+use kernel::hil::time::Alarm;
+use kernel::Chip;
+use rv32i;
+use rv32i::csr::{mcause, mie::mie, mip::mip, CSR};
+use rv32i::pmp::PMP;
+
+use crate::interrupts;
+use crate::plic::Plic;
+use crate::plic::PLIC;
+use kernel::InterruptService;
+
+type QemuRv32VirtPMP = PMP<4>;
```

Comment:
QEMU has 16 PMP regions

Reply:
Changed accordingly. Can you tell me where that is defined though? I couldn't figure it out, at some point literally grepped for `16` in the (slightly outdated) QEMU source tree.

Reply:
`MAX_RISCV_PMPS` is what you are looking for. It's in `cpu.h`